### PR TITLE
Issue #21 develop client

### LINKED ISSUE #21: organize and clean up client.js
The client.js file should have a nearly identical API to the SocketServer.

### PR DESCRIPTION
**Pull request by crookse**
Closes #21

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -1,54 +1,141 @@
...
+   * @return {SocketClient}
+   */
   constructor(options) {
     this.configs = {
       hostname: options.hostname || "localhost",
       port: options.port || "3000",
     };
-    this.listening = {};
+    this.decoder = new TextDecoder();
+    this.listening_to = {};
```
Comment (saragee3) [inline client.js:38]: or `receiving` ?

File: client.js [hunk 2]
```diff
@@ -1,54 +1,141 @@
...
+   *     The name of the channel or event.
+   * @param {Function} callback
+   *     The callback to execute on receipt of a message from the channel or event.
+   */
+  on(channelOrEvent, callback) {
+    if (this.connection.readyState === 1) {
+      if (!this.listening_to[channelOrEvent]) {
+        this.listening_to[channelOrEvent] = null;
+      }
+      this.listening_to[channelOrEvent] = callback;
```
Comment (saragee3) [inline client.js:65]: this should prob be an array actually in the case where multiple areas want to receive from the same channel?

File: client.js [hunk 3]
```diff
@@ -1,54 +1,141 @@
...
+      const message = JSON.stringify({ listening_to: channelOrEvent });
```
Comment (saragee3) [inline client.js:66]: actually maybe `listening_to` makes sense... you want to listen to a channel and ask to _be_ a receiver lol

File: client.js (hunk 2 above)
Comment (crookse) [reply client.js:65]: making a question issue about this

File: client.js (hunk 1 above)
Comment (crookse) [reply client.js:38]: making an issue about this

**Pull request merged**